agent/api/task/task.go

```diff
@@ -2408,20 +2408,14 @@ func (task *Task) dockerPortMap(container *apicontainer.Container, dynamicHostPo
 			// we are guaranteed to have the entire hostPortRange in a single network binding while sending this info to ECS.
 			hostPortRange, err := getHostPortRange(numberOfPorts, protocol, dynamicHostPortRange)
 			if err != nil {
-				// in the odd case where we're unable to find a contiguous set of host ports, we fall back to docker dynamic port
-				// assignment for the requested ContainerPortRange.
-				logger.Error("Unable to find contiguous host ports for container, falling back to "+
-					"docker dynamic port assignment", logger.Fields{
+				logger.Error("Unable to find contiguous host ports for container", logger.Fields{
 					field.TaskID:         task.GetID(),
 					field.Container:      container.Name,
 					"containerPortRange": containerPortRange,
 					field.Error:          err,
 				})
-
-				// append individual container port from the containerPortRange into the containerPortSet.
-				// this will ensure that we populate network bindings for ports that docker dynamically assigned.
-				for port := startContainerPort; port <= endContainerPort; port++ {
-					containerPortSet[port] = struct{}{}
+				if err != nil {
```

Comment:
the parent if condition and this one is the same? so we can log and then return immediately?

Reply:
Thanks for catching this! Line 2417 is redundant, and should be removed.